common/sagas/transaction/network/gas.ts

```diff
@@ -48,6 +50,18 @@ export function* shouldEstimateGas(): SagaIterator {
       continue;
     }
 
+    // If we have a hard-coded limit, use it immediately and skip estimate
+    const msg = action.type === TypeKeys.TO_FIELD_SET && getAddressMessage(action.payload.raw);
```

Comment:
This way of overriding gas limit could be unreliable as some actions cause a casade of actions to be fired, which would mean `TypeKeys.TO_FIELD_SET` wouldn't be the action type, but the gas limit would still be re-estimated, wiping out the override.

A better way of implementing this is running this in a seperate saga which watches over the `toField` of the state. Then when the field is changed and matches (e.g. getAddressMessage returns a valid value). 

0) Store the previous state of what the auto gas estimation variable was in the saga
1) turn of auto gas estimation
2) grab the messsage gas limit
3) Dispatch the custom gas limit (now it wont be overriden because auto gas is off)

While steps 0-3 are running, if the to field changes by anything:
1) set auto gas estimation back to its original value
2) Run steps 0-3 again

If the state is cleared / transaction is sent (they should be under `reset` action)
1) set auto gas estimation back to its original value

Also, it may be a good idea to warn the user that you're no longer auto estimating, and instead its a predefined value.

Reply:
That's a good point about gas limit estimation possibly being kicked off by other actions. However, I'd like to leave gas estimate on and still give the users the visual that gas limit is uneditable & auto-calculated in these instances. To me, the gas estimation checkbox means "figure it out for me", not literally estimate it via the node. If we know better, I think we should leave the field disabled and just set it for them. And if they want to change it themselves, unchecking the checkbox is still an option (Which would override this code path, since the early exit for offline or estimate off happens before it.)

So I think the only change I want to make here is to open it up to all actions, not just TO_FIELD_SET. Let me know if you think there was anything else important that this misses.

Reply:
That works too, I'd still split this into another saga though and call it from within this one just for maintainability / testability sake. You'll also need to modify how you get the message (probably by using a selector to get the address instead) so that it works on any gas estimation related actions.

Reply:
Made a selector for address messages, busted it out into its own function, and added unit tests. Lemme know if there's anything else!